pandas_flavor/register.py

```diff
@@ -1,5 +1,85 @@
 from functools import wraps
-from pandas.api.extensions import register_series_accessor, register_dataframe_accessor
+from pandas.api.extensions import (
+    register_series_accessor,
+    register_dataframe_accessor,
+)
+import inspect
+from contextlib import nullcontext
+
+method_call_ctx_factory = None
+
+
+def handle_pandas_extention_call(method, method_signature, obj, args, kwargs):
+    """
+    This function is called when the user calls the registered method on pandas dataframe object.
+    pandas extention mechanism passes args and kwargs of original method call as it was applied to obj
+
+    Implementation uses global var *method_call_ctx_factory"
```

Comment:
Improved grammar and switched to using Markdown syntax.

```suggestion
    Our implementation uses the global variable `method_call_ctx_factory`.
```